modules/sd_samplers_common.py

```diff
@@ -22,21 +22,26 @@ def setup_img2img_steps(p, steps=None):
     return steps, t_enc
 
 
-approximation_indexes = {"Full": 0, "Approx NN": 1, "Approx cheap": 2}
+approximation_indexes = {"Full": 0, "Tiny AE": 1, "Approx NN": 2, "Approx cheap": 3}
 
 
 def single_sample_to_image(sample, approximation=None):
     if approximation is None:
         approximation = approximation_indexes.get(opts.show_progress_type, 0)
 
-    if approximation == 2:
-        x_sample = sd_vae_approx.cheap_approximation(sample)
-    elif approximation == 1:
-        x_sample = sd_vae_approx.model()(sample.to(devices.device, devices.dtype).unsqueeze(0))[0].detach()
+    if approximation == 1:
+        x_sample = sd_vae_taesd.decode()(sample.to(devices.device, devices.dtype).unsqueeze(0))[0].detach()
+        x_sample = sd_vae_taesd.TAESD.unscale_latents(x_sample)
+        x_sample = torch.clamp((x_sample * 0.25) + 0.5, 0, 1)
     else:
-        x_sample = processing.decode_first_stage(shared.sd_model, sample.unsqueeze(0))[0]
+        if approximation == 3:
+            x_sample = sd_vae_approx.cheap_approximation(sample)
+        elif approximation == 2:
+            x_sample = sd_vae_approx.model()(sample.to(devices.device, devices.dtype).unsqueeze(0))[0].detach()
+        else:
+            x_sample = processing.decode_first_stage(shared.sd_model, sample.unsqueeze(0))[0]
+        x_sample = torch.clamp((x_sample + 1.0) / 2.0, min=0.0, max=1.0)

```

Comment:
Why nested and reordered value checks?  Seems cleaner to keep the original structure like:
```
if approximation == 3:
elif approximation == 2:
elif approximation == 1:
else:
```

Reply:
You ignored line 43, it's not suitable for all cases, layering the `if` is necessary.

Reply:
Ah of course! Sorry about my mistake